clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java

```diff
@@ -177,4 +184,70 @@ public void clear() throws TimeoutException {
       throw new ServerStoreProxyException(e);
     }
   }
+
+  @Override
+  public Iterator<Chain> iterator() throws TimeoutException {
+    EhcacheEntityResponse.IteratorBatch iteratorBatch = openIterator();
+    if (iteratorBatch.isTerminated()) {
+      return iteratorBatch.getChains().iterator();
+    } else {
+      UUID iteratorId = iteratorBatch.getIdentity();
+      return new Iterator<Chain>() {
+
+        private boolean terminalBatch = false;
+        private Iterator<Chain> batch = iteratorBatch.getChains().iterator();
+
+        @Override
+        public boolean hasNext() {
+          return !terminalBatch || batch.hasNext();
```

Comment:
Hmmm ... If you're in the last batch, then you certainly need to ask `batch` if it has more items.  But, if you aren't in the last batch, then you may have consumed the last item in the penultimate batch and need to ask the next batch if _it_ has any items -- it may or may not -- you don't know until you receive the last batch and see if it actually has any items.

Reply:
Last batch can't be empty... by contract with the server.